.gitmodules

```diff
@@ -7,3 +7,7 @@
 [submodule "dlpack"]
 	path = dlpack
 	url = https://github.com/dmlc/dlpack
+[submodule "glad"]
```

Comment:
What is glad used for, is it possible to only use OpenGL standard and not rely on new libraries?

Reply:
We actually rely on 2 libraries.

- `glad`: This is the library to help you use the correct version of OpenGL.
  For example, suppose your OS supports OpenGL 3.3, then the OS is able to give you any OpenGL version below 3.3 on your request. However, if you just `#include <GL/gl.h>` or whatever header your OS provides, the function prototypes are for OpenGL 1.3 or something, so you can't use OpenGL 3.3 even your OS supports it. The OS provides a way of querying the function pointers of OpenGL 3.3 APIs. That's what `glad` does. It is possible that we don't rely on `glad`, but with a huge cost - we must write our own code to get all the function pointers.

- glfw: This is the library to help you create an OpenGL context in a cross-platform way.
  You can't use any OpenGL API before creating a context. GLFW wraps the context creation code of different OS'es to provide a unified API. It is possible that we don't rely on `glfw`, but with a huge cost - we must write context initialization code for different platforms.

I can remove the dependencies on those libraries, but that can only happen after Dec 12 or something. We are focusing on producing a final-report-able version currently.

Reply:
Sure, I think these are fine as long as we fix this at the final merge.

Reply:
AFAIK GLFW itself wraps up all the functions. Maybe GLAD is not needed since GLFW is already there. If my memory went wrong please inform me.